Combat/Source/Points.cpp

```diff
@@ -0,0 +1,111 @@
+#include "Points.hpp"
+#include <cstdlib>
+
+Points::Points(const uint maxPoints)
+{
+   _maxPoints = maxPoints;
+   _currentPoints = maxPoints;
+}
+
+Points::Points(const uint maxPoints, const uint currentPoints)
+{
+      _maxPoints = maxPoints;
+
+      if(currentPoints > maxPoints)
+      {
+         _currentPoints = maxPoints;
+      }
+      else
+      {
+         _currentPoints = currentPoints;
+      }
+}
+
+// Sprawdzanie na karcie czy currentMP >= amount
+void Points::remove(const uint amount)
+{
+   if(amount > _currentPoints)
+   {
+      _currentPoints = 0;
+   }
+   else
+   {
+      _currentPoints -= amount;
+   }
+}
+
+void Points::add(const uint amount)
+{
+   if(_currentPoints + amount > _maxPoints)
+   {
+      _currentPoints = _maxPoints;
+   }
+   else
+   {
+      _currentPoints += amount;
+   }
+}
+
+void Points::setMaxTo (const uint amount)
+{
+   _maxPoints = amount;
+}
+
+void Points::resetCurrent()
+{
+   _currentPoints = _maxPoints;
+}
+
+//Operators
+bool operator == (const Points& lhs, const Points& rhs)
+{
+   return lhs._currentPoints == rhs._currentPoints;
+}
+
+bool operator != (const Points& lhs, const Points& rhs)
+{
+   return !(lhs == rhs);
+}
+
+bool operator > (const Points& lhs, const Points& rhs)
+{
+   return lhs._currentPoints > rhs._currentPoints;
+}
+
+bool operator >= (const Points& lhs, const Points& rhs)
+{
+   return lhs._currentPoints >= rhs._currentPoints;
+}
+
+bool operator < (const Points& lhs, const Points& rhs)
+{
+   return !(lhs >= rhs);
+}
+
+bool operator <= (const Points& lhs, const Points& rhs)
+{
+   return !(lhs > rhs);
+}
+
+Points operator + (const Points& lhs, const Points& rhs)
```

Comment:
Wszędzie wyżej trzymacie się, że edytujecie tylko _currentPoints. Dla operator- i operator+ robicie też dla maxa, chyba nie jest to spójne ;) Moim zdaniem lepiej tylko _currentPoints.

Reply:
Jak dokładnie miałoby działać dodawanie? Załóżmy, że są dwie instancje Points: 
p1 z currentPoints = 40 i maxPoints = 50 oraz
p2 z currentPoints = 30 i maxPoints = 60.
W programie jest `auto result = p1 + p2;` Jakie wartości mają mieć atrybuty obiektu result?
Czy maxPoints ma być takie jak u obiektu z lewej czy z prawej strony? Czy zawsze ma wybierać wyższą lub niższą wartość? A może  po prostu maxPoints ma być równe sumie currentPoints obu obiektów?